standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/StatisticsManagementTask.java

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.common.TableName;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.DeleteColumnStatisticsRequest;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.api.hive_metastoreConstants;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.hadoop.hive.metastore.conf.TimeValidator;
+import org.apache.hadoop.hive.metastore.model.MPartitionColumnStatistics;
+import org.apache.hadoop.hive.metastore.model.MTable;
+import org.apache.hadoop.hive.metastore.model.MTableColumnStatistics;
+import org.apache.hadoop.hive.metastore.utils.MetaStoreUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import javax.jdo.PersistenceManager;
+import javax.jdo.Query;
+
+/**
+ * Statistics management task is primarily responsible for auto deletion of table column stats based on a certain frequency
+ *
+ * If some table column statistics are older than the period time, they should be deleted automatically
+ * Statistics Retention - If "partition.retention.period" table property is set with retention interval, when this
+ * metastore task runs periodically, it will drop partitions with age (creation time) greater than retention period.
+ * Dropping partitions after retention period will also delete the data in that partition.
+ *
+ */
+public class StatisticsManagementTask implements MetastoreTaskThread {
+    private static final Logger LOG = LoggerFactory.getLogger(StatisticsManagementTask.class);
+
+    // global
+    public static final String STATISTICS_AUTO_DELETION = "statistics.auto.deletion";
+    public static final String STATISTICS_RETENTION_PERIOD = "statistics.retention.period";
+
+    // The 2 configs for users to set in the conf
+    // this is an optional table property, if this property does not exist for a table, then it is not excluded
+    public static final String STATISTICS_AUTO_DELETION_EXCLUDE_TBLPROPERTY = "statistics.auto.deletion.exclude";
+
+    private static final Lock lock = new ReentrantLock();
+
+    // these are just for testing
+    private static int completedAttempts;
+    private static int skippedAttempts;
```

Comment:
nit: remove these two. They are not being used.

Reply:
Acknowledged.